### ISSUE TITLE
Error while deleting entry from completed builds list

### ISSUE THREAD

**Issue by Yaon-C2H8N2** [docker/for-mac#7202]
### Description

In the "Builds" tab on the desktop app, some completed builds entry can't be deleted from the displayed list in the desktop application even after multiple deletion.

All those remaining build have been deleted multiple times but are still displayed in the list :

![image](https://github.com/docker/for-mac/assets/73293479/713277bb-edc4-4b35-abed-6641b275e388)

Clicking on any of them then brings this this error :

![image](https://github.com/docker/for-mac/assets/73293479/d6fff894-f9c1-4722-a960-c464d9be352d)

### Reproduce

1. Delete a completed build
2. Some aren't deleted (can't really identify any pattern)

### Expected behavior

The deleted entry should not be displayed in the list.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.10
 Version:           25.0.1
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        29cf629
 Built:             Tue Jan 23 23:06:12 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.27.0 (135262)
 Engine:
  Version:          25.0.1
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       71fa3ab
  Built:            Tue Jan 23 23:09:35 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.27
  GitCommit:        a1496014c916f9e62104b33d1bb5bd03b0858e59
 runc:
  Version:          1.1.11
  GitCommit:        v1.1.11-0-g4bccb38
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    25.0.1
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     /Users/yaon/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.3-desktop.1
    Path:     /Users/yaon/.docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.22
    Path:     /Users/yaon/.docker/cli-plugins/docker-debug
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/yaon/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     /Users/yaon/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     /Users/yaon/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.0
    Path:     /Users/yaon/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/yaon/.docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.3.0
    Path:     /Users/yaon/.docker/cli-plugins/docker-scout
WARNING: Plugin "/Users/yaon/.docker/cli-plugins/docker-scan" is not valid: failed to fetch metadata: fork/exec /Users/yaon/.docker/cli-plugins/docker-scan: no such file or directory

Server:
 Containers: 5
  Running: 2
  Paused: 0
  Stopped: 3
 Images: 9
 Server Version: 25.0.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: a1496014c916f9e62104b33d1bb5bd03b0858e59
 runc version: v1.1.11-0-g4bccb38
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.6.12-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 3.832GiB
 Name: docker-desktop
 ID: a662d072-dc07-4348-ab6c-385f48df7fda
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Diagnostics ID

6F8EADFA-9068-408A-9B1B-096B92C7893E/20240226213823

**Comment by app-o-matix** [comment]
I've had the same issue. I couldn't manually delete two builds with the same name. Hard to tell if quitting and restarting Docker Desktop helps, because the results of being able to delete builds doing so is inconsistent. I've done bulk deletes of three builds with the same name where it only deletes the latest one.

I quit out of Docker Desktop and ran these terminal commands in an attempt to delete them and these are the results:
```
~/ -> docker system df
TYPE            TOTAL     ACTIVE    SIZE      RECLAIMABLE
Images          0         0         0B        0B
Containers      0         0         0B        0B
Local Volumes   0         0         0B        0B
Build Cache     0         0         0B        0B
~/ -> docker builder prune --all --force
Total:	0B
~/ -> docker container prune --force
Total reclaimed space: 0B
~/ -> docker buildx du
Reclaimable:	0B
Total:		0B
~/ ->
```
So it would seem nothing was affected by these commands, that there was nothing to delete. When I reopened Docker Desktop, the builds were still there, but this time I was able to manually delete them. Whether quitting and restarting had an affect, or running these commands had an affect, or some combination, I don't know. Perhaps just coincidence and it is something else. But that is the behavior I'm seeing.
